### PR TITLE
Lodestar validator with Lighthouse Bn integration

### PR DESCRIPTION
**Pull request by mpetrunic**

### CODE REVIEW

File: src/renderer/services/eth2/client/lighthouse/validator.ts [hunk 1]
```diff
@@ -27,9 +27,34 @@ export class LighthouseValidatorApiClient implements IValidatorApi {
     private readonly client: HttpClient;
     private readonly config: IBeaconConfig;
     private readonly beaconApi: IBeaconApi;
-    
+
     public constructor(options: IBeaconClientOptions, beaconApi: IBeaconApi) {
-        this.client = new HttpClient(options.baseUrl, {axios: {transformResponse: bigIntParse}});
+        this.client = new HttpClient(options.baseUrl, {
+            axios: {
```
Comment (morrigan) [inline src/renderer/services/eth2/client/lighthouse/validator.ts:33]: maybe move to some requests utils file

File: src/renderer/services/validator/interop_keys.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import {PrivateKey} from "@chainsafe/bls";
+import {toBufferBE} from "bigint-buffer";
+import {bytesToBigInt, intToBytes} from "@chainsafe/lodestar-utils";
+import {hash} from "@chainsafe/ssz";
+
+const CURVE_ORDER = BigInt("52435875175126190479447740508185965837690552500527637822603658699938581184513");
+
+export function getInteropKey(index: number): PrivateKey {
```
Comment (morrigan) [inline src/renderer/services/validator/interop_keys.ts:8]: What's this? Sounds like it should be a some part of lodestar

File: src/renderer/services/eth2/client/lighthouse/beacon.ts [hunk 3]
```diff
@@ -17,18 +17,46 @@ import {ILighthouseSyncResponse} from "./types";
...
+                            return bigIntParse(data);
+                        } catch (e) {
+                            return data;
+                        }
+                    }
+                ]
+            }
+        });
         this.config = options.config;
+        this.logger = options.logger;
```
Comment (morrigan) [inline src/renderer/services/eth2/client/lighthouse/beacon.ts:57]: Should be some function to retrieve logger

File: README.md [hunk 4]
```diff
@@ -51,7 +51,12 @@ By typing `docker-compose up` in terminal, you will get:
 - geth eth1 node with private POA network
 - deposit contract deployed on that private eth1 network
     - contract is always deployed on address `0x2F1598e74b146F5687174C13f8EDCF490B2492e3`
-- prysmatic eth2 private network
+- lighthouse eth2 private network
```
Comment (morrigan) [inline README.md:54]: Add instructions for `docker-compose`

Comment (morrigan) [reply README.md:54]: Also for creating account
1. importing private key 
2. skip withdrawal
3. configure beacon node: Network localhost, node localhost:5052

File: .gitignore [hunk 5]
```diff
@@ -9,4 +9,5 @@ coverage
 *.log
 package-lock.json
 *.db
-/cg_testnet/genesis.ssz
+/eth2_testnet/genesis.ssz
```
[the remark this answers is not in the document]
Comment (mpetrunic) [reply .gitignore:12]: Think so, but let's change it in separate PR as it's not really important I guess

File: src/renderer/services/validator/interop_keys.ts (hunk 2 above)
Comment (mpetrunic) [reply src/renderer/services/validator/interop_keys.ts:8]: It some convention to generate determinitic keys that lighthouse is using. Lodestar is using PrivateKey.fromInt() which is not the same

File: src/renderer/services/eth2/client/lighthouse/beacon.ts (hunk 3 above)
Comment (mpetrunic) [reply src/renderer/services/eth2/client/lighthouse/beacon.ts:57]: why?

Comment (morrigan) [reply src/renderer/services/eth2/client/lighthouse/beacon.ts:57]: These ones that should be displayed as well to the user?

Comment (mpetrunic) [reply src/renderer/services/eth2/client/lighthouse/beacon.ts:57]: but you are passing your implementation of this logger interface which pushes logs to stream

**Pull request merged**